miner/worker.go

```diff
@@ -400,7 +400,7 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 		case head := <-w.chainHeadCh:
 			clearPending(head.Block.NumberU64())
 			timestamp = time.Now().Unix()
-			// commit(false, commitInterruptNewHead)
+			commit(true, commitInterruptNewHead)
```

Comment:
any quick way to understand the effect of `true`/`false` here?

originally it's `false`, right? why we want to change it to `true` here

Reply:
This involves multiple changes, and we need to use transaction-driven block generation, where true means no new blocks are generated when there are no transactions in the pool.
The problem with the previous change is that when the volume exceeds the upper limit, new blocks are no longer generated.
The easiest way to do this is based on the Ethereum source code by simply changing 'commit(false,' to true.

Reply:
> any quick way to understand the effect of `true`/`false` here?
> 
> originally it's `false`, right? why we want to change it to `true` here

Line403 is the key change.

Reply:
I see, so basically the controlling "no new blocks are generated when there are no transactions in the pool" logic is here: https://github.com/scroll-tech/go-ethereum/blob/e9a5a982d65077d2bd4ad59bf67ed53c6b285ee1/miner/worker.go#L1012-L1016

Reply:
Yes, that right.